Source/SwiftLintFramework/Rules/Style/ExplicitSelfRule.swift

```diff
@@ -0,0 +1,203 @@
+            }
+            """
+        ],
+        requiresFileOnDisk: true
+    )
+
+    public func validate(file: File, compilerArguments: [String]) -> [StyleViolation] {
+        return violationRanges(in: file, compilerArguments: compilerArguments).map {
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severity,
+                           location: Location(file: file, characterOffset: $0.location))
+        }
+    }
+
+    public func correct(file: File, compilerArguments: [String]) -> [Correction] {
+        let violations = violationRanges(in: file, compilerArguments: compilerArguments)
+        let matches = file.ruleEnabled(violatingRanges: violations, for: self)
+        if matches.isEmpty { return [] }
+
+        var contents = file.contents.bridge()
+        let description = type(of: self).description
+        var corrections = [Correction]()
+        for range in matches.reversed() {
+            contents = contents.replacingCharacters(in: range, with: "self.").bridge()
+            let location = Location(file: file, characterOffset: range.location)
+            corrections.append(Correction(ruleDescription: description, location: location))
+        }
+        file.write(contents.bridge())
+        return corrections
+    }
+
+    private func violationRanges(in file: File, compilerArguments: [String]) -> [NSRange] {
+        guard !compilerArguments.isEmpty else {
+            queuedPrintError("""
+                Attempted to lint file at path '\(file.path ?? "...")' with the \
+                \(type(of: self).description.identifier) rule without any compiler arguments.
+                """)
+            return []
+        }
+
+        let allCursorInfo: [[String: SourceKitRepresentable]]
+        do {
+            let byteOffsets = try binaryOffsets(file: file, compilerArguments: compilerArguments)
+            allCursorInfo = try file.allCursorInfo(compilerArguments: compilerArguments,
+                                                   atByteOffsets: byteOffsets)
+        } catch {
+            queuedPrintError(String(describing: error))
+            return []
+        }
+
+        let cursorsMissingExplicitSelf = allCursorInfo.filter { cursorInfo in
+            guard let kindString = cursorInfo["key.kind"] as? String else { return false }
+            return kindsToFind.contains(kindString)
+        }
+
+        guard !cursorsMissingExplicitSelf.isEmpty else {
+            return []
+        }
+
+        let contents = file.contents.bridge()
+
+        return cursorsMissingExplicitSelf.compactMap { cursorInfo in
+            guard let byteOffset = cursorInfo["swiftlint.offset"] as? Int64 else {
+                queuedPrintError("couldn't convert offsets")
+                return nil
+            }
+
+            return contents.byteRangeToNSRange(start: Int(byteOffset), length: 0)
+        }
+    }
+}
+
+private let kindsToFind = Set([
```

Comment:
you could avoid the runtime creation from an array: 

```swift
private let kindsToFind: Set = [
// ...
]
```

Reply:
Done